pkg/apis/kubermatic/v1/policy_template.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2025 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// PolicyTemplateResourceName represents "Resource" defined in Kubernetes.
+	PolicyTemplateResourceName = "policytemplates"
+
+	// PolicyTemplateKind represents "Kind" defined in Kubernetes.
+	PolicyTemplateKind = "PolicyTemplate"
+)
+
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Enforced",type=boolean,JSONPath=".spec.enforced",description="Whether the policy is mandatory"
+
+// PolicyTemplate defines a reusable blueprint of a Kyverno policy.
+type PolicyTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PolicyTemplateSpec `json:"spec,omitempty"`
+}
+
+type PolicyTemplateSpec struct {
+	// Title is the title of the policy, specified as an annotation in the Kyverno policy
+	Title string `json:"title"`
+
+	// Description is the description of the policy, specified as an annotation in the Kyverno policy
+	Description string `json:"description"`
+
+	// Category is the category of the policy, specified as an annotation in the Kyverno policy
+	//
+	// +optional
+	Category string `json:"category,omitempty"`
+
+	// Severity indicates the severity level of the policy
+	//
+	// +optional
+	Severity string `json:"severity,omitempty"`
+
+	// Scope specifies the scope of the policy.
+	// Can be one of: global, project, or cluster
+	//
+	// +kubebuilder:validation:Enum=global;project;cluster
+	Scope string `json:"scope"`
```

Comment:
While reviewing I've been confused by the name of this field, thinking it's also scoping `PolicyTemplate`. Turns of this field is needed for UI visibility at different access levels for different KKP actors. Probably other people will also be confused by this name, let's change it to something like `UIVisibility` or anything else you can thing of, but `Scope`.